src/effects/RealtimeEffectStateUI.cpp

```diff
@@ -0,0 +1,160 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  RealtimeEffectStateUI.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+#include "RealtimeEffectStateUI.h"
+
+#include <cassert>
+
+#include "EffectUI.h"
+#include "RealtimeEffectState.h"
+
+#include "EffectManager.h"
+#include "ProjectWindow.h"
+#include "Track.h"
+
+namespace
+{
+const RealtimeEffectState::RegisteredFactory realtimeEffectStateUIFactory { [](auto& state)
+   { return std::make_unique<RealtimeEffectStateUI>(state); }
+};
+} // namespace
+
+
+RealtimeEffectStateUI::RealtimeEffectStateUI(RealtimeEffectState& state)
+    : mRealtimeEffectState(state)
+{
+}
+
+RealtimeEffectStateUI::~RealtimeEffectStateUI()
+{
+   Hide();
+}
+
+bool RealtimeEffectStateUI::IsShown() const noexcept
+{
+   return mEffectUIHost != nullptr;
+}
+
+void RealtimeEffectStateUI::Show(AudacityProject& project)
+{
+   if (mEffectUIHost != nullptr && mEffectUIHost->IsShown())
+   {
+      // Bring the host to front
+      mEffectUIHost->Raise();
+      return;
+   }
+
+   const auto ID = mRealtimeEffectState.GetID();
+   const auto effectPlugin = EffectManager::Get().GetEffect(ID);
+
+   if (effectPlugin == nullptr)
+      return;
+
+   EffectUIClientInterface* client = effectPlugin->GetEffectUIClientInterface();
+
+   // Effect has no client interface
+   if (client == nullptr)
+      return;
+
+   // mUIHost will acquire the correct instance
```

Comment:
copy-pasted comment refers to a nonexistent variable

Reply:
Comment is trying to explain why `release()` doesn't leak below.  It's because the dialog has a non null parent.

Reply:
Well, that means it's a bad comment anyway :-) It should be about `EffectInstance` lifetime

Reply:
It is no longer needed on the other hand, the (weak) pointer to EffectInstance is no longer stored